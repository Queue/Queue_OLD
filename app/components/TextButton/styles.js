//
// TextButton styles

import { StyleSheet } from 'react-native';
import Colors from '../../lib/colors';

export default styles = StyleSheet.create({
  textButton: {
    padding: 5,
    letterSpacing: 1,
    paddingBottom: 15,
    fontSize: 14,
    color: Colors.primaryForeground,
    textAlign: 'center'
  }
});
