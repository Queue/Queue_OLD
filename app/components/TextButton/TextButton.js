//
// TextButton component

import React from 'react';
import { TouchableHighlight, Text } from 'react-native';
import styles from './styles';
import Colors from '../../lib/colors';
import Fonts from '../../lib/fonts';

export default TextButton = ({font, press, text, size}) => {

  // Set font size defaults to 14
  styles.size = {
    fontSize: size,
    fontFamily: font
  }

  return (
    <TouchableHighlight
      onPress = {press}
      underlayColor = {'transparent'}>
      <Text style = {[styles.textButton, styles.size]}>{text}</Text>
    </TouchableHighlight>
  );
};

TextButton.propTypes = {
  press: React.PropTypes.func,
  text: React.PropTypes.string,
  size: React.PropTypes.number,
  font: React.PropTypes.string
};

TextButton.defaultProps = {
  text: 'Add Text',
  size: 14,
  font: Fonts.content
}
