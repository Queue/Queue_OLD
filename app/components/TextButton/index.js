//
// TextButton index

import TextButton from './TextButton.js';
export default TextButton;
