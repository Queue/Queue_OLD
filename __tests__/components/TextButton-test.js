//
// TextButton component test

import React from 'react';
import TextButton from '../../app/components/TextButton';
import renderer from 'react-test-renderer';

let testFunc = () => {
  console.warn('');
};

it('renders correctly', () => {
  const tree = renderer.create(
    <TextButton
      press={testFunc}
      text={'Log In'}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
