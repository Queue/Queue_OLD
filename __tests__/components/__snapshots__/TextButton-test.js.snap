exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#48C948",
          "fontSize": 14,
          "letterSpacing": 1,
          "padding": 5,
          "paddingBottom": 15,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "Helvetica Neue",
          "fontSize": 14,
        },
      ]
    }>
    Log In
  </Text>
</View>
`;
